config/context.go

```diff
@@ -0,0 +1,73 @@
+package config
+
+import (
+	"context"
+
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+
+	"github.com/gruntwork-io/terragrunt/config/hclparse"
+	"github.com/gruntwork-io/terragrunt/options"
+)
+
+// Context provides various variables that are used throughout all funcs and passed from function to function.
+// Using `Context` makes the code more readable..
+type Context struct {
```

Comment:
NIT: Could we use a clearer name? This one conflicts with the `Context` used in Go for canceling goroutines and the like... Whereas here, this feels more like `ParsingContext` or similar...

Reply:
Fair point. Replaced, thanks!